src/Orleans/Serialization/BuiltInTypes.cs

```diff
@@ -1637,7 +1637,7 @@ internal static object CopyDateTimeOffset(object obj)
 
         internal static void SerializeType(object obj, BinaryTokenStreamWriter stream, Type expected)
         {
-            stream.Write(((Type)obj).OrleansTypeName());
+            stream.Write(((Type)obj).OrleansTypeKeyString());
```

Comment:
This won't break backward compatibility with data serialized and persisted previously, right?